### ISSUE TITLE
Slack.py engine 100% CPU load due to missing time.sleep(1)

### ISSUE THREAD

**Issue by basdusee** [saltstack/salt#38517]
### Description of the Issue/Bug 
The slack.py engine in Salt will fry the CPU at 100% (or at least one core) due to a missing time.sleep(1) statement at the end of the main "while True:" loop in the main "start" function of the slack.py engine

Adding a "time.sleep(1)" indented with 12 spaces at line no. 227 solves the issue.
This regards /usr/lib/python2.7/dist-packages/salt/engines/slack.py on my local install.

### Setup
Here are the relevant parts in /etc/salt/master:
```
engines:
  - slack:
      token: 'xoxb-supersecret'
      control: True
      valid_users:
        - saltuser
      valid_commands:
        - test.ping
        - cmd.run
```

### Steps to Reproduce Issue
Configure the slack engine with above config (or similar) and start salt-master

### Versions Report
```
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-57-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by gtmanfred** [comment]
Thanks for opening a pull request for this!
Daniel

**Comment by cachedout** [comment]
Closed via #38520
